analysis/src/CompletionFrontEnd.ml

```diff
@@ -1324,10 +1322,29 @@ let completionWithParser1 ~currentFile ~debug ~offset ~path ~posCursor
                         inJsx = !inJsxContext;
                       }))
             | None -> ())
-        | Pexp_apply {funct = {pexp_desc = Pexp_ident compName}; args}
-          when Res_parsetree_viewer.is_jsx_expression expr ->
+        | Pexp_jsx_element
+            ( Jsx_unary_element
+                {
+                  jsx_unary_element_tag_name = compName;
+                  jsx_unary_element_props = props;
+                }
+            | Jsx_container_element
+                {
+                  jsx_container_element_tag_name_start = compName;
+                  jsx_container_element_props = props;
+                } ) ->
           inJsxContext := true;
```

Comment:
I'm curious why do we need `inJsxContext` flag since we are already in JSX branch? Maybe we can document this decision somewhere in the code.

Reply:
This is used in CompletionBackEnd to autocomplete `->React.string` when dotting into a string. `CPField` and `CPPipe` do have a comment for it.